dev-lang/rust/rust-1.21.0.recipe

```diff
@@ -0,0 +1,211 @@
+SOURCE_URI_4="https://github.com/rust-lang/hoedown/archive/$srcGitRev4.tar.gz"
+SOURCE_URI_5="https://github.com/rust-lang/libc/archive/$srcGitRev5.tar.gz"
+SOURCE_URI_6="https://github.com/rust-lang/llvm/archive/$srcGitRev6.tar.gz"
+SOURCE_URI_7="https://github.com/rust-lang/cargo/archive/$cargoVersion.tar.gz"
+SOURCE_URI_8="https://github.com/rust-lang-nursery/nomicon/archive/$srcGitRev8.tar.gz"
+SOURCE_URI_9="https://github.com/rust-lang-nursery/reference/archive/$srcGitRev9.tar.gz"
+SOURCE_URI_10="https://github.com/rust-lang/book/archive/$srcGitRev10.tar.gz"
+SOURCE_URI_11="https://github.com/rust-lang-nursery/rls/archive/$srcGitRev11.tar.gz"
+SOURCE_URI_12="https://github.com/rust-lang-nursery/compiler-builtins/archive/$srcGitRev12.tar.gz"
+SOURCE_FILENAME="rust-$srcGitRev.tar.gz"
+CHECKSUM_SHA256="a0e674cce176a5c90804d394e57887b278acfcdbda4fe0f2ae5876531a94faa4"
+CHECKSUM_SHA256_2="ac595c03dd5dcd3ad66eee76bec2e95770259e4aa4db5df875c2bbcf7781a441"
+CHECKSUM_SHA256_3="dfcce5a555e0aff5f7ba7681968b77529c806c3109ede78eb1b45a0c4455db4b"
+CHECKSUM_SHA256_4="623889a326754e93a4a7a8a3487699ea786f4ca9b118f7468c12f8b9966905e6"
+CHECKSUM_SHA256_5="e40ca6485049911c319b717404181675e88c2d56d2d8d27395cc85b0e4f428fb"
+CHECKSUM_SHA256_6="e5481733ae30cc985674b7494e48ba8717748af9265514dd8516ceceb32ae8f0"
+CHECKSUM_SHA256_7="6740e2f8d035a5a5246a12e84a141d37ad4c2b826355aff497e6f1fcfb637aef"
+CHECKSUM_SHA256_8="d3e255dfe63cabd342229a53605cd94ae6df912e4771f88a1a4de7af3a59d8f2"
+CHECKSUM_SHA256_9="7e0fb24af053ac9415568b01442a5ab491a9ed5b841e9441808100d63175cb68"
+CHECKSUM_SHA256_10="e2059881a339efff63e94d363b75bf34dfad2a0297bc46c568ba40dcc71f2d6f"
+CHECKSUM_SHA256_11="fa74c46f239ebd1557dea184de3caf7d76c003c6de64aa7fc07c2c9d9f0543f3"
+CHECKSUM_SHA256_12="82fa7ada1bdadbba9f780847dc4f9340f15e5f137d838c98ca7e30ec3dda6684"
+SOURCE_DIR="rust-$srcGitRev"
+PATCHES="rust-$portVersion.patchset"
+REVISION="1"
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	rust$secondaryArchSuffix = $portVersion
+	cmd:rustc = $portVersion
+	cmd:rustdoc = $portVersion
+	cmd:rust_gdb = $portVersion
+	cmd:rust_lldb = $portVersion
+	cmd:cargo$secondaryArchSuffix = $cargoVersion
+	cmd:rls = $rlsVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcurl$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:cmp
+	cmd:file
+	cmd:find
+	cmd:git
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:python
+	cmd:sed
+	cmd:tar
+	cmd:which
+	cmd:xargs
+	rust_bin$secondaryArchSuffix = $portVersion
+	"
+CONFLICTS="
+	rust_bin$secondaryArchSuffix
```

Comment:
Wouldn't it be nicer to remove the `CONFLICTS` in rust non-bin and add the following one to rust_bin?
```
CONFLICTS="
	rust$secondaryArchSuffix
	"
```

(BTW, the prefered place for the `CONFLICTS` definition is just after `REQUIRES`.)

BTW, if replacing the `rust_bin$secondaryArchSuffix = $cargoVersion` (in `BUILD_REQUIRES`) by `cmd:cargo$secondaryArchSuffix = $cargoVersion` works fine then it might be a good idea to do so, as this would allow to build rust with rust itself or with rust_bin.
Hmm, I might be missing something, so this is really just an untested suggestion :)